consensus/types/src/beacon_state.rs

```diff
@@ -1071,28 +1097,38 @@ impl<E: EthSpec> BeaconState<E> {
             return Err(Error::SlotOutOfBounds);
         }
 
-        let seed = self.get_beacon_proposer_seed(slot, spec)?;
-        let indices = self.get_active_validator_indices(epoch, spec)?;
+        if let Ok(proposer_lookahead) = self.proposer_lookahead() {
+            // Post-Fulu
+            let index = slot.as_usize().safe_rem(E::slots_per_epoch() as usize)?;
```

Comment:
This calculation only works if the `slot` is in the current epoch, which we check above. Although I wonder if eventually we might want to loosen that restriction so we can take advantage of the cache to access proposers in future epochs.

Maybe just worth adding a comment here to that effect.

Reply:
done!